File: src/common/django/shared/errors/bad_request_error.py
from typing import List
from .custom_error import CustomError, IErrorStruct


class BadRequestError(CustomError):
    status_code = 400

    def __init__(self, message: str, field: str = None):
        self.message = message
        self.field = field
        super().__init__(self.message)

    def serialize_errors(self) -> List[IErrorStruct]:
        return [IErrorStruct(message=self.message, field=self.field).to_dict()]

File: src/common/nodejs/src/errors/bad-request-error.ts
import { CustomError, IErrorStruct } from './custom-error';

export class BadRequestError extends CustomError {
  statusCode = 400;

  constructor(public message: string = 'Bad Request', public field: string = 'None') {
    super(message);
    Object.setPrototypeOf(this, BadRequestError.prototype);
  }

  serializeErrors(): IErrorStruct[] {
    return [{ message: this.message, field: this.field }];
  }
}
